src/awkward/_v2/contents/regulararray.py

```diff
@@ -33,12 +33,15 @@ def __init__(
                     type(self).__name__, repr(content)
                 )
             )
-        if not (ak._util.isint(size) and size >= 0):
-            raise TypeError(
-                "{0} 'size' must be a non-negative integer, not {1}".format(
-                    type(self).__name__, size
+        if not isinstance(size, ak._v2._typetracer.UnknownLengthType):
+            if not (ak._util.isint(size) and size >= 0):
+                raise TypeError(
+                    "{0} 'size' must be a non-negative integer, not {1}".format(
+                        type(self).__name__, size
+                    )
```

Comment:
It's okay for this RegularArray to have an unknown `size` because it's a temporary RegularArray. The final Form will have a known `size`.